packages/berlin/src/components/tabs/TabsHeader.tsx

```diff
@@ -1,21 +1,20 @@
-import { Fragment, useState } from 'react';
+import { Fragment } from 'react';
 import { Body } from '../typography/Body.styled';
 import { Tab } from './TabsHeader.styled';
 
 type TabsHeaderProps = {
   tabNames: string[];
-  initialTab?: string;
+  activeTab: string;
   className?: string;
   onTabChange?: (tab: string) => void;
 };
 
-export function TabsHeader({ tabNames, initialTab, className, onTabChange }: TabsHeaderProps) {
-  const [activeTab, setActiveTab] = useState<string>(initialTab || tabNames[0]);
-
+export function TabsHeader({ tabNames, activeTab, className, onTabChange }: TabsHeaderProps) {
   const handleTabClick = (tab: string) => {
-    setActiveTab(tab);
-    if (onTabChange) {
-      onTabChange(tab);
+    if (activeTab !== tab) {
+      if (onTabChange) {
+        onTabChange(tab);
+      }
     }
   };
```

Comment:
i think initialTab is better as a prop because it actually will only set the tab sent on mount and will not change if it was already mounted

Reply:
if we want to change the component because the prop has changed, we can either add a prop key to re render the whole component or add logic like:

```
const [prevItems, setPrevItems] = useState(items);
  if (items !== prevItems) {
    setPrevItems(items);
    setSelection(null);
  }

```

read https://react.dev/learn/you-might-not-need-an-effect  `Resetting all state when a prop changes` and `Adjusting some state when a prop changes`